src/FixedDataTableCellGroup.js

```diff
@@ -65,57 +65,79 @@ class FixedDataTableCellGroupImpl extends React.Component {
 
   componentWillMount() {
     this._initialRender = true;
+    this._staticCellArray = [];
   }
 
   componentDidMount() {
     this._initialRender = false;
   }
 
   render() /*object*/ {
-    var props = this.props;
-    var columns = props.columns;
-    var cells = new Array(columns.length);
-    var contentWidth = sumPropWidths(columns);
+    const props = this.props;
+    const columnsToRender = props.columnsToRender;
+    const columns = props.columns;
+    const contentWidth = sumPropWidths(columns);
 
-    var isColumnReordering = props.isColumnReordering && columns.reduce(function (acc, column) {
+    const isColumnReordering = props.isColumnReordering && columns.reduce(function (acc, column) {
       return acc || props.columnReorderingData.columnKey === column.props.columnKey;
     }, false);
 
-    var currentPosition = 0;
-    for (var i = 0, j = columns.length; i < j; i++) {
-      var columnProps = columns[i].props;
-      var cellTemplate = columns[i].template;
-      var recyclable = columnProps.allowCellsRecycling && !isColumnReordering;
-      if (!recyclable || (
-        currentPosition - props.left <= props.width &&
-        currentPosition - props.left + columnProps.width >= 0)) {
-        var key = columnProps.columnKey || 'cell_' + i;
-        cells[i] = this._renderCell(
-          props.rowIndex,
-          props.rowHeight,
-          columnProps,
-          cellTemplate,
-          currentPosition,
-          key,
-          contentWidth,
-          isColumnReordering
-        );
+    if (this.props.isScrolling) {
+      // We are scrolling, so there's no need to display any cells which lie outside the viewport.
+      // We still need to render them though, so as to not cause any unmounts.
+      this._staticCellArray.forEach((cell, i) => {
+        this._staticCellArray[i] = React.cloneElement(this._staticCellArray[i], { visible: false });
+      });
+    } else {
+      // reset the static cell array if scrolling is stopped
+      // this is done so that only cells inside the viewport are considered for vertical scrolling
+      this._staticCellArray = [];
```

Comment:
I think this single line is worth a discussion.

Let's say there are 1000 columns, and each column's width increases from left to right.
So initially the viewport might only render 5 columns, and at the end it might need to render 100.

As we move towards right, the no: of columns in the viewport increases. This will cause mounts to occur.
Now when we reach the end, there will be a total of 95 mounts that happened, and 100 columns will be rendered for each vertical scroll. This is expected behavior.

Now if we go back to the beginning, only 5 columns need to be rendered. But the staticCellArray still contains an extra of 95 columns from the previous renders. This means, on each vertical render, all the 100 columns are considered for render again. 

I did add a `shouldComponentUpdate` logic to skip the other 95, but I think the DOM will be smaller and updates will be faster if we just eliminate the components when scrolling has stopped. This is what's done here.

So while scrolling, there will be mounts happening. And at the time of unscrolling, there will be unmounts.

Reply:
Interesting.  It may be worth trying to profile / time a case like this and see whether shouldComponentUpdate checks or unmounts would be faster.